Dockerfile

```diff
@@ -0,0 +1,39 @@
+# The MIT License
+#
+#  Copyright 2020 Loghi Perinpanayagam.
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to deal
+#  in the Software without restriction, including without limitation the rights
+#  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#  copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+#  THE SOFTWARE.
+
+
+FROM jenkins/agent
+
+# if we want to install via apt
+USER root
+
+RUN apt-get update
+RUN apt-get install python -y
+RUN apt-get install python-pip -y
+
+RUN pip install ipython
+RUN pip install ipykernel
+RUN pip install jupyter-client
+RUN pip install protobuf
+RUN pip install grpcio
```

Comment:
@Loghijiaha each `RUN` command creates a new layer in the Docker file system. So it's better if we can group some of these comments. Here's an example:

https://github.com/kinow/cylc-docker/blob/7d6a165bd0b4f1c12bb5c411267e34462cfa00f7/cylc-flow/8.0a1/Dockerfile#L34

For the rationale and more about best practices: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/